pkg/lambda/grpc/transport.go

```diff
@@ -10,15 +12,79 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 
 	pbtransport "github.com/conductorone/baton-sdk/pb/c1/transport/v1"
+	"google.golang.org/protobuf/reflect/protoregistry"
 )
 
 type Request struct {
 	msg *pbtransport.Request
 }
 
+/*
+UnmarshalJSON unmarshals the JSON into a Request of course,
+
+filtering out any annotations that are not known to the global registry
+which happens frequently for new features and would otherwise require
+rolling every lambda function.
+
+Our requests are small in size, dwarfed by the work of the connector,
+so the performance impact is negligible.
+*/
 func (f *Request) UnmarshalJSON(b []byte) error {
 	f.msg = &pbtransport.Request{}
-	return protojson.Unmarshal(b, f.msg)
+
+	// Parse top-level as raw and surgically filter req.annotations
+	var top map[string]json.RawMessage
+	if err := json.Unmarshal(b, &top); err != nil {
+		return err
+	}
+	// Track if we actually modified the payload.
+	changed := false
+
+	if reqRaw, ok := top["req"]; ok && len(reqRaw) > 0 {
+		var reqObj map[string]json.RawMessage
+		if err := json.Unmarshal(reqRaw, &reqObj); err == nil {
+			if annsRaw, ok := reqObj["annotations"]; ok && len(annsRaw) > 0 {
+				var anns []json.RawMessage
+				if err := json.Unmarshal(annsRaw, &anns); err == nil {
+					wellKnownAnnotations := make([]json.RawMessage, 0, len(anns))
+					for _, ann := range anns {
+						var annObj map[string]json.RawMessage
+						if err := json.Unmarshal(ann, &annObj); err != nil {
+							continue
+						}
+						var typeURL string
+						if t, ok := annObj["@type"]; ok {
+							if err := json.Unmarshal(t, &typeURL); err == nil {
+								// It would be nice to log here, but we have no context.
+								if _, err := protoregistry.GlobalTypes.FindMessageByURL(typeURL); err == nil {
+									wellKnownAnnotations = append(wellKnownAnnotations, ann) // keep only known types
```

Comment:
A different approach could be to package the unknown annotation into a known type, so we can log or do something with it later?